evm/src/cpu/kernel/interpreter.rs

```diff
@@ -1129,14 +1165,20 @@ impl<'a> Interpreter<'a> {
         Ok(())
     }
 
-    fn jump_to(&mut self, offset: usize) -> anyhow::Result<(), ProgramError> {
-        // The JUMPDEST rule is not enforced in kernel mode.
-        if !self.is_kernel() && self.jumpdests.binary_search(&offset).is_err() {
-            return Err(ProgramError::InvalidJumpDestination);
-        }
-
+    fn jump_to(&mut self, offset: usize, is_jumpi: bool) -> anyhow::Result<(), ProgramError> {
         self.generation_state.registers.program_counter = offset;
 
+        if offset == KERNEL.global_labels["observe_new_address"] {
```

Comment:
I believe this is only for debugging purposes, but I thought it might also be useful to store them on the interpreter side.